gen/crocs_user_export

```diff
@@ -8,42 +8,42 @@ use JSON::XS;
 
 our $SERVICE_NAME = "crocs_user_export";
 our $PROTOCOL_VERSION = "3.0.0";
-my $SCRIPT_VERSION = "3.0.0";
+my $SCRIPT_VERSION = "3.0.1";
 
 perunServicesInit::init;
 my $DIRECTORY = perunServicesInit::getDirectory;
-my $data = perunServicesInit::getHierarchicalData;
+my $data = perunServicesInit::getHashedHierarchicalData;
 
 #Constants
 our $A_USER_FIRSTNAME;            *A_USER_FIRSTNAME =        \'urn:perun:user:attribute-def:core:firstName';
 our $A_USER_LASTNAME;             *A_USER_LASTNAME =         \'urn:perun:user:attribute-def:core:lastName';
 our $A_USER_LOGIN;                *A_USER_LOGIN=             \'urn:perun:user:attribute-def:def:login-namespace:mu';
 
-my %attributesByLogin;
+my $attributesByLogin;
 
-my @resourcesData = $data->getChildElements;
-foreach my $rData (@resourcesData) {
-	my @membersData = $rData->getChildElements;
-	foreach my $mData (@membersData) {
-		my %memberAttributes = attributesToHash $mData->getAttributes;
-		$attributesByLogin{$memberAttributes{$A_USER_LOGIN}} = \%memberAttributes;
-	}
+foreach my $memberId ($data->getMemberIdsForFacility()) {
+	my $login = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN );
+
+	$attributesByLogin->{$login}->{$A_USER_FIRSTNAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_FIRSTNAME );
+	$attributesByLogin->{$login}->{$A_USER_LASTNAME} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LASTNAME );
+	$attributesByLogin->{$login}->{$A_USER_LOGIN} = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_LOGIN );
 }
 
 
 my @users;
-for my $login (sort keys %attributesByLogin) {
-	my $values = $attributesByLogin{$login};
+for my $login (sort keys %{$attributesByLogin}) {
+	my $values = $attributesByLogin->{$login};
 	my $user = {};
 	$user->{"firstname"} = $values->{$A_USER_FIRSTNAME} || "";
 	$user->{"lastname"}  = $values->{$A_USER_LASTNAME} || "";
-	$user->{"UCO"}       = $values->{$A_USER_LOGIN};
+	$user->{"UCO"}       = $values>{$A_USER_LOGIN};
```

Comment:
Missing `-` in `$values>{$A_USER_LOGIN};`, but when I think about it, we can simplify it further. We can use simple `$user->{"UCO"} = $login`.
Because we already have value for login as a key in the main hash. There is no need to store it also under the `$A_USER_LOGIN` key within the stored user object.

Reply:
Yes, this is the way I wanted to do it in the first commit, but I did a mistake (I placed $values instead of $login). So I agree and I simplified it.